### PR TITLE
[vim] tidy up coding styles for vim/syntax/swift.vim

### PR DESCRIPTION
**Pull request by tokorom**
**Update only utils/vim/syntax/swift.vim**

The keywords and the regular expressions were on different lines on most lines. I thought they were more readable and conflict-resistant.
Therefore, I'd like to match all the lines to that style. Just by modifying the styles, I have not changed the content at all.

I would like to update these before creating other pull requests.

### CODE REVIEW

File: utils/vim/syntax/swift.vim [hunk 1]
```diff
@@ -141,40 +141,59 @@ syn match swiftTypeDeclaration skipwhite skipempty nextgroup=swiftType,swiftInOu
...
+      \ /[+\-]\?\<\([0-9][0-9_]*\)\([.][0-9_]*\)\?\([eE][+\-]\?[0-9][0-9_]*\)\?\>/
+syn match swiftHex
+      \ /[+\-]\?\<0x[0-9A-Fa-f][0-9A-Fa-f_]*\(\([.][0-9A-Fa-f_]*\)\?[pP][+\-]\?[0-9][0-9_]*\)\?\>/
+syn match swiftOct
+      \ /[+\-]\?\<0o[0-7][0-7_]*\>/
+syn match swiftBin
+      \ /[+\-]\?\<0b[01][01_]*\>/
+
+syn match swiftOperator skipwhite skipempty nextgroup=swiftTypeParameters
+      \ "\.\@<!\.\.\.\@!\|[/=\-+*%<>!&|^~]\@<!\(/[/*]\@![/=\-+*%<>!&|^~]*\|*/\@![/=\-+*%<>!&|^~]*\|->\@![/=\-+*%<>!&|^~]*\|[=+%<>!&|^~][/=\-+*%<>!&|^~]*\)"
```
Comment (tokorom) [inline utils/vim/syntax/swift.vim]: I changed + to " because the syntax highlight for this file got strange after this line.

**Comment by compnerd** [comment]
I hadn't done something like that to try to keep the history as reasonable as possible.  I'm not opposed to that.  The quote character change however, I think is a great idea.

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by CodaFi** [comment]
I'm going to rebase this and merge it in #28330.

**Pull request closed without merging**